File: 1233RemoveSubFolders.py
class Solution(object):
    def removeSubfolders(self, folder):
        """
        :type folder: List[str]
        :rtype: List[str]
        """

        folder.sort()

        ans = [folder[0]]

        for i in range(1, len(folder)):
            lastFolder = ans[len(ans) - 1] + "/"
            
            if not folder[i].startswith(lastFolder):
                ans.append(folder[i])
    

        return ans

        
solution = Solution()

folder = ["/a/b", "/a", "/c/d", "/c/d/e", "/c/f"]
answer = solution.removeSubfolders(folder)

print(answer)

File: 1233RemoveSubFolders.js
/**
 * @param {string[]} folder
 * @return {string[]}
 */
const removeSubfolders = (folder) => {
  folder.sort();

  const ans = [folder[0]];

  for (let i = 1; i < folder.length; i++) {
    const lastFolder = ans[ans.length - 1] + "/";

    if (!folder[i].startsWith(lastFolder)) {
      ans.push(folder[i]);
    }
  }

  return ans
};

// removeSubfolders(["/a", "/a/b", "/c/d", "/c/d/e", "/c/f"]);
// removeSubfolders(["/a","/a/b/c","/a/b/d"])
// removeSubfolders(["/a/b/c","/a/b/ca","/a/b/d"])
// removeSubfolders(["/ab","/abc","/abcd"])
